ui/jobs/components/brd.ts

```diff
@@ -1,162 +1,197 @@
+
+    this.straightShotProc = this.bars.addProcBox({
+      id: 'brd-procs-straightshotready',
+      fgColor: 'brd-color-straightshotready',
+      threshold: 1000,
+    });
+    this.straightShotProc.bigatzero = false;
+
+    // DoT
+    this.causticBiteBox = this.bars.addProcBox({
+      id: 'brd-procs-causticbite',
+      fgColor: 'brd-color-causticbite',
+      notifyWhenExpired: true,
+    });
+    this.stormBiteBox = this.bars.addProcBox({
+      id: 'brd-procs-stormbite',
+      fgColor: 'brd-color-stormbite',
+      notifyWhenExpired: true,
+    });
+
+    // Song
+    this.songBox = this.bars.addProcBox({
+      id: 'brd-procs-song',
+      fgColor: 'brd-color-song',
+    });
+    this.repertoireBox = this.bars.addResourceBox({
+      classList: ['brd-color-song'],
+    });
+    this.repertoireTimer = this.bars.addTimerBar({
+      id: 'brd-timers-repertoire',
+      fgColor: 'brd-color-song',
+    });
+    this.repertoireTimer.toward = 'right';
+    this.repertoireTimer.stylefill = 'fill';
+    // Only with-DoT-target you last attacked will trigger bars timer.
+    // So it work not well in multiple targets fight.
+    this.repertoireTracker = new DotTracker({ emitter: this.emitter, player: this.player });
+    this.repertoireTracker.onTick([
+      EffectId.Stormbite,
+      EffectId.Windbite,
+      EffectId.CausticBite,
+      EffectId.VenomousBite,
+    ], () => {
+      this.repertoireTimer.duration = kDoTTickInterval;
+    });
+    this.soulVoiceBox = this.bars.addResourceBox({
+      classList: ['brd-color-soulvoice'],
+    });
+  }
+
+  override onMobGainsEffectFromYou(id: string): void {
+    // Iron jaws just refreshes these effects by gain once more,
+    // so it doesn't need to be handled separately.
+    // Log line of getting DoT comes a little late after DoT appear on target,
+    // so -0.5s
+    switch (id) {
+      case EffectId.Stormbite:
+      case EffectId.Windbite:
+        this.stormBiteBox.duration = 30 - 0.5;
+        break;
+
+      case EffectId.CausticBite:
+      case EffectId.VenomousBite:
+        this.causticBiteBox.duration = 30 - 0.5;
+        break;
+    }
+  }
+
+  override onJobDetailUpdate(jobDetail: JobDetail['BRD']): void {
+    this.songBox.fg = computeBackgroundColorFrom(this.songBox, 'brd-color-song');
+    this.repertoireBox.parentNode.classList.remove('minuet', 'ballad', 'paeon', 'full');
+    this.repertoireBox.innerText = '';
     if (jobDetail.songName === 'Minuet') {
-      repertoireBox.innerText = jobDetail.songProcs.toString();
-      repertoireBox.parentNode.classList.add('minuet');
-      songBox.fg = computeBackgroundColorFrom(songBox, 'brd-color-song.minuet');
-      songBox.threshold = 5;
-      repertoireBox.parentNode.classList.remove('full');
+      this.repertoireBox.innerText = jobDetail.songProcs.toString();
+      this.repertoireBox.parentNode.classList.add('minuet');
+      this.songBox.fg = computeBackgroundColorFrom(this.songBox, 'brd-color-song.minuet');
```

Comment:
I don't see this box ever proc.  However, this also happens without your patch, so it's not an error here, but a bug for later.